tests/src/runtimeApi/stream/hipStreamAddCallback.cpp

```diff
@@ -23,41 +23,65 @@ THE SOFTWARE.
  * HIT_END
  */
 
-// Test under-development. Call hipStreamAddCallback function and see if it works as expected.
-
+#include <stdio.h>
 #include "hip/hip_runtime.h"
 #include "test_common.h"
 
 #ifdef __HIP_PLATFORM_HCC__ 
 #define HIPRT_CB
 #endif
 
-class CallbackClass
+__global__ void vector_square(float *C_d, float *A_d, size_t N)
 {
-public:
-    static void HIPRT_CB Callback(hipStream_t stream, hipError_t status, void *userData);
+    size_t offset = (blockIdx.x * blockDim.x + threadIdx.x);
+    size_t stride = blockDim.x * gridDim.x ;
+
+    for (size_t i=offset; i<N; i+=stride) {
+        C_d[i] = A_d[i] * A_d[i];
+    }
+}
 
-private:
-    void callbackFunc(hipError_t status);
-};
+float *A_h, *C_h;
+size_t N = 1000000;
 
-void HIPRT_CB CallbackClass::Callback(hipStream_t stream, hipError_t status, void *userData)
+static void HIPRT_CB Callback(hipStream_t stream, hipError_t status, void *userData)
 {
-    CallbackClass* obj = (CallbackClass*) userData;
-    obj->callbackFunc(status);
+    for (size_t i=0; i<N; i++) {
+        if (C_h[i] != A_h[i] * A_h[i]) {
+            HIPCHECK(hipErrorUnknown);
+        }
+    }
+    printf ("PASSED!\n");
 }
 
-void CallbackClass::callbackFunc(hipError_t status)
+int main(int argc, char *argv[])
 {
-     HIPASSERT(status==hipSuccess);
-}
+    float *A_d, *C_d;
+    size_t Nbytes = N * sizeof(float);
+
+    A_h = (float*)malloc(Nbytes);
+    HIPCHECK(A_h == 0 ? hipErrorMemoryAllocation : hipSuccess );
+    C_h = (float*)malloc(Nbytes);
+    HIPCHECK(C_h == 0 ? hipErrorMemoryAllocation : hipSuccess );
+
+    // Fill with Phi + i
+    for (size_t i=0; i<N; i++) 
```

Comment:
++i

Reply:
Array index starts at 0. So i++ is correct.